routers/user/auth.go

```diff
@@ -648,6 +625,13 @@ func LinkAccountPostRegister(ctx *context.Context, cpt *captcha.Captcha, form au
 	}
 	log.Trace("Account created: %s", u.Name)
 
+	// Link the oAuth account to the user
+	err := models.LinkAccountToUser(u, gothUser.(goth.User))
```

Comment:
Now that 1.1.0 is out yes, this is a problem. Too bad you didn't want to merge this before (we saw people had problems with early upgrades anyway, so this would have not been different)